### ISSUE TITLE
RVX patches crashes when built with rvmm

### ISSUE THREAD

**Issue by Fred-Vatin** [j-hc/revanced-magisk-module#545]
Hi, tried to install RVX youtube and music using this config with last commit 9b44264 from December 9:
```toml
enable-magisk-update = true   # set this to false if you do not want to receive updates for the module in magisk app

# add 'enabled = false' for not patching a specific app or remove it from the config
# see https://github.com/j-hc/revanced-magisk-module/blob/main/CONFIG.md for more detailed explanations

# you can use rvmm-config-gen to generate a config
# GLOBAL
rv-brand = "Revanced"

[Music-EX]
app-name = "Music"
patches-source = "inotia00/revanced-patches"
cli-source = "inotia00/revanced-cli"
build-mode = "both"
arch = "both"
apkmirror-dlurl = "https://www.apkmirror.com/apk/google-inc/youtube-music"

[YouTube-EX]
app-name = "YouTube"
patches-source = "inotia00/revanced-patches"
cli-source = "inotia00/revanced-cli"
build-mode = "both"
apkmirror-dlurl = "https://www.apkmirror.com/apk/google-inc/youtube"
excluded-patches = "'Theme' 'Alternative thumbnails' 'Enable debugging'"
```

Apps crash immediately. I cleaned up data via android settings and some revanced related files remained in `data/adb` after module deletion before reinstalling modules to no avail.

ROOT apk built with RVX manager work.

**Comment by j-hc** [comment]
> ROOT apk built with RVX manager work.

**Comment by Kevinr99089** [comment]
Same problem with non-root builds
Application does not work if patched with builder, but works with RVX Manager

**Comment by soudblox** [comment]
Are there any temporary fixes for this?

**Comment by Pocarai** [comment]
> Are there any temporary fixes for this?

Or either permanent solution since we use this feature on Obtainium to make the app updated frequently

**Comment by Pocarai** [comment]
How crazy that I just discovered this feature and then after auto update in non-root it'll crash unexpected happend.

**Comment by jill88** [comment]
same for me using the zip for rooted rvx YouTube in magisk on Pixel 8pro.  app crashes immediately.

**Comment by ftugit** [comment]
x6 pro critical error, won't start

**Comment by Quasar-0330** [comment]
Redmi 12 5G (crDroid v11 beta) dont launch

**Comment by JCVargas029** [comment]
Latest Revance build crashing.

**Comment by ChaseTheCode** [comment]
I can confirm from texting on several rooted devices that it affects them all. I also tried clearing cache and storage without any luck.

**Comment by JCVargas029** [comment]
Why my comment is deleted by @ghost. It is relevant to the topic! Weird.

**Comment by toolhas4degrees** [comment]
In the rvx changelog it says new one creates warnings that must be deleted. Don't know if this is the cause

**Comment by Kevinr99089** [comment]
The latest commit doesn't seem to have solved the crash problem with the application.

**Comment by ChaseTheCode** [comment]
> The latest commit doesn't seem to have solved the crash problem with the application.

I can confirm after testing on several devices today after the patch, and clearing cache and storage for the app that the latest update did not resolve the issue.

**Comment by E85Addict** [comment]
I've been doing some testing and I've concluded what works and what doesn't but I'm not sure why it is. Using the repo to make the extended modules clearly doesn't work like everyone here has stated, they just don't launch. I've tried using the CLI manually in Ubuntu using WSL2, Java Zulu 17 jdk installed with both the official, yours and inotia00's CLI and they all result the exact same crash/failure to launch. So I manually copied all the files after running the repo over to my windows directory, I also have java zulu 17 jdk installed on windows and I manually ran the CLI through the windows side on windows 11 and for whatever reason any builds using Windows and the CLI manually just WORK including modules. So from my testing using the inotia00's patches on Ubuntu/Github Actions using Ubuntu with Java Zulu 17/21 both produce crashing apks/module. Doing it on Windows 11 using the same version of Java and patches they just work. What would cause this is beyond me but that's what I've gathered.

**Comment by j-hc** [comment]
> I've been doing some testing and I've concluded what works and what doesn't but I'm not sure why it is. Using the repo to make the extended modules clearly doesn't work like everyone here has stated, they just don't launch. I've tried using the CLI manually in Ubuntu using WSL2, Java Zulu 17 jdk installed with both the official, yours and inotia00's CLI and they all result the exact same crash/failure to launch. So I manually copied all the files after running the repo over to my windows directory, I also have java zulu 17 jdk installed on windows and I manually ran the CLI through the windows side on windows 11 and for whatever reason any builds using Windows and the CLI manually just WORK including modules. So from my testing using the inotia00's patches on Ubuntu/Github Actions using Ubuntu with Java Zulu 17/21 both produce crashing apks/module. Doing it on Windows 11 using the same version of Java and patches they just work. What would cause this is beyond me but that's what I've gathered.

i already know the reason. developer of rv extended matches made it so if its built with my repo, it crashes. i can fix it but im not sure if i want to play this cat and mouse game.

**Comment by peternmuller** [comment]
> i already know the reason. developer of rv extended patches made it so if its built with my repo, it crashes. i can fix it but im not sure if i want to play this cat and mouse game.

@j-hc, as @hldr4 mentionned:
> Doesn't appear to be a deliberate breaking change, moreso a compatibility conflict.
You might want to try to fix it once, and then see if the developer of rv extended patches will break it again, so that you know if it's really deliberate or not.

**Comment by Fred-Vatin** [comment]
> developer of rv extended patches made it so if its built with my repo

Is there anything we can report to inotiaa since the problem seems to be there ?

**Comment by j-hc** [comment]
fixed d439382
